addon/services/scheduler.js

```diff
@@ -71,12 +45,11 @@ const Scheduler = Service.extend({
     const queue = this.queues[queueName];
     queue.isActive = false;
 
-    for (let i = 0; i < queue.tasks.length; i += 2) {
-      const callback = queue.tasks[i];
-      const token = queue.tasks[i+1];
+    while (queue.size() > 0) {
```

Comment:
I believe this should likely be a `flush` method on `Queue`...

Reply:
Agree, but want to land this in small increments. I can convert now but was trying to make the diff smaller...